File: test-suite-kotlin/src/test/kotlin/io/micronaut/security/docs/authentication/RunAsService.kt
package io.micronaut.security.docs.authentication

import io.micronaut.context.annotation.Requires
import io.micronaut.security.annotation.RunAs
import io.micronaut.security.annotation.RunAs.Attribute
import io.micronaut.security.authentication.Authentication
import io.micronaut.security.context.SecurityContextHolder
import jakarta.inject.Singleton

@Requires(property = "spec.name", value = "RunAsTest")
//tag::clazz[]
@RunAs(
    name = "aegon",
    roles = ["TARGARYEN"],
    attributes = [
        Attribute(key = "family_name", value = "Targaryen"),
        Attribute(key = "given_name", value = "Aegon")
    ]
)
@Singleton
open class RunAsService {
    open fun changeAuth(): Authentication? = SecurityContextHolder.getSecurityContext().authentication
}
//end::clazz[]

File: test-suite-python/src/test/python/micronaut/security/docs/authentication/RunAsService.py
from jakarta.inject import Singleton
from micronaut.context.annotation import Requires
from micronaut.security.annotation import RunAs
from micronaut.security.authentication import Authentication
from micronaut.security.context import SecurityContextHolder


@Requires(property="spec.name", value="RunAsTest")
# tag::clazz[]
@RunAs(
    name="aegon",
    roles=["TARGARYEN"],
    attributes=[
        RunAs.Attribute(key="family_name", value="Targaryen"),
        RunAs.Attribute(key="given_name", value="Aegon"),
    ],
)
@Singleton
class RunAsService:
    def change_auth(self) -> Authentication:
        return SecurityContextHolder.getSecurityContext().getAuthentication()
# end::clazz[]
